server/src/server/server_future.rs

```diff
@@ -215,6 +242,29 @@ impl<T: RequestHandler> ServerFuture<T> {
         Ok(())
     }
 
+    /// Register a TlsListener to the Server. The TlsListener should already be bound to either an
+    /// IPv6 or an IPv4 address.
+    ///
+    /// To make the server more resilient to DOS issues, there is a timeout. Care should be taken
+    ///  to not make this too low depending on use cases.
+    ///
+    /// # Arguments
+    /// * `listener` - a bound TCP (needs to be on a different port from standard TCP connections) socket
+    /// * `timeout` - timeout duration of incoming requests, any connection that does not send
+    ///               requests within this time period will be closed. In the future it should be
+    ///               possible to create long-lived queries, but these should be from trusted sources
+    ///               only, this would require some type of whitelisting.
+    /// * `pkcs12` - certificate used to announce to clients
+    #[cfg(feature = "tls")]
```

Comment:
I think this ends up only working with OpenSSL. In the server library that’s effectively the same thing right now, but to be correct, this should be “dns-over-openssl”